src/include/daos_fs.h

```diff
@@ -443,6 +443,26 @@ dfs_stat(dfs_t *dfs, dfs_obj_t *parent, const char *name,
 int
 dfs_ostat(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf);
 
+#define DFS_SET_ATTR_MODE	(1 << 0)
+#define DFS_SET_ATTR_ATIME	(1 << 1)
+#define DFS_SET_ATTR_MTIME	(1 << 2)
+/**
+ * set stat attributes for a file and fetch new values.  If the object is a
+ * symlink the link itself is modified.  See dfs_stat() for which entries
+ * are filled.
+ *
+ * \param[in]	dfs	Pointer to the mounted file system.
+ * \param[in]	obj	Open object (File, dir or syml) to modify.
+ * \param[in,out]
+ *		stbuf	[in]: Stat struct with the members set.
+ *			[out]: Stat struct with all valid members filled.
+ * \param[in]	flags	Bitmask of flags to set
+ *
+ * \return		0 on Success. Negative errno on Failure.
+ */
+int
+dfs_osetstat(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags);
```

Comment:
I'd rather have this be just a set rather a set and get (see comments in the function).
Does dfuse need dfs_osetattr()? or can this be a setattr with a parent and a name?

Reply:
I'm fine with changing it to a set only although that removes our ability to handle a race condition.  This does need to be dfs_osetattr() as if it was a setattr we'd need to pass in a oid to verify as part of the operation which brings up back to the same place as osetattr(), but with a worse API.